runtime/lua/vim/filetype/detect.lua

```diff
@@ -1,15 +1,28 @@
+-- Contains filetype detection functions converted to Lua from Vim's autoload/runtime/dist/ft.vim file.
+
+-- Here are a few guidelines to follow when porting a new function:
+-- Sort the function alphabetically and omit 'ft' or 'check' from the new function name.
+-- Use :find instead of :match / :sub if possible.
+-- When '=~' is used to match a pattern, there are two possibilities:
+-- If the pattern only contains lowercase characters treat the comparison as case-insensitive.
+-- Otherwise, treat it as case-sensitive. Basically, we apply 'smartcase' (if upper case
+-- characters are used in the original pattern, then it's likely that case does matter).
+
+-- When '\k', '\<' or '\>' is used in a pattern, use the matchre function.
+-- Note that vim.regex is case-sensitive by default, so add the '\c' flag if only
+-- lowercase letters are present in the pattern:
+-- Example:
+-- `if line =~ '^\s*unwind_protect\>'` => `if matchre(line, [[\c^\s*unwind_protect\>]])
```

Comment:
```suggestion
--  * Sort the function alphabetically and omit 'ft' or 'check' from the new function name.
--  * Use ':find' instead of ':match' / ':sub' if possible.
--  * When '=~' is used to match a pattern, there are two possibilities:
--     - If the pattern only contains lowercase characters, treat the comparison as case-insensitive.
--     - Otherwise, treat it as case-sensitive. 
--     (Basically, we apply 'smartcase': if upper case characters are used in the original pattern, then 
--     it's likely that case does matter).
--  * When '\k', '\<' or '\>' is used in a pattern, use the 'matchregex' function.
--     Note that vim.regex is case-sensitive by default, so add the '\c' flag if only lowercase letters 
--     are present in the pattern:
--     Example:
--     `if line =~ '^\s*unwind_protect\>'` => `if matchregex(line, [[\c^\s*unwind_protect\>]])
```

Reply:
Done